examples/dbscan/ArborX_DetailsDBSCANVerification.hpp

```diff
@@ -123,14 +123,22 @@ bool verifyBoundaryPointsConnectToCorePoints(ExecutionSpace const &exec_space,
 // Check that cluster indices are unique
 template <typename ExecutionSpace, typename IndicesView, typename OffsetView,
           typename ClusterView>
-bool verifyClustersAreUnique(ExecutionSpace const &, IndicesView indices,
-                             OffsetView offset, ClusterView clusters,
-                             int core_min_size)
+bool verifyClustersAreUnique(ExecutionSpace const &exec_space,
+                             IndicesView indices, OffsetView offset,
+                             ClusterView clusters, int core_min_size)
 {
   int n = clusters.size();
 
-  auto clusters_host =
-      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, clusters);
+  // FIXME we don't want to modify the clusters view in this check. What we
+  // want here is to create a view on the host, and deep_copy into it.
+  // create_mirror_view_and_copy won't work, because it is a no-op if clusters
+  // is already on the host.
+  decltype(Kokkos::create_mirror_view(Kokkos::HostSpace{},
+                                      std::declval<ClusterView>()))
+      clusters_host(Kokkos::ViewAllocateWithoutInitializing(
+                        "ArborX::DBSCAN::clusters_host"),
+                    clusters.size());
+  Kokkos::deep_copy(exec_space, clusters_host, clusters);
```

Comment:
Would be useful to have in `Kokkos`.

Reply:
Sure would be nice if some kind soul could introduce it there. Definitely not doing it myself.